File: old_trunk/cost/iona/orbacus/cpp/poa/TestUtil.cpp
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <fstream>

using namespace CORBA;
using namespace PortableServer;

POA_ptr
TestUtil_GetRootPOA(ORB_ptr orb)
{
    Object_var obj;

    try
    {
        obj = orb -> resolve_initial_references("RootPOA");
    }
    catch(const ORB::InvalidName&)
    {
	OB_ERROR("Error: can't resolve `RootPOA'");
	exit(1);
    }

    if(is_nil(obj))
    {
	OB_ERROR("Error: `RootPOA' is a nil object reference");
	exit(1);
    }

    POA_var root = POA::_narrow(obj);

    if(is_nil(root))
    {
	OB_ERROR("Error: `RootPOA' is not a POA object reference");
	exit(1);
    }

    return root._retn();
}

bool
TestUtil_Compare(const ObjectId& id1, const ObjectId& id2)
{
    return (id1.length() == id2.length() &&
            memcmp(id1.get_buffer(), id2.get_buffer(), id1.length()) == 0);
}

File: old_trunk/cost/iona/orbacus/java/poa/TestUtil.java
// **********************************************************************
//
// Copyright (c) 2000
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package test.poa;

import org.omg.CORBA.*;
import org.omg.PortableServer.*;

final class TestUtil
{
    static POA
    GetRootPOA(ORB orb)
    {
	org.omg.CORBA.Object obj = null;

	try
	{
	    obj = orb.resolve_initial_references("RootPOA");
	}
	catch(org.omg.CORBA.ORBPackage.InvalidName ex)
	{
	    System.err.println("Error: can't resolve `RootPOA'");
	    System.exit(1);
	}

	if(obj == null)
	{
	    System.err.println("Error: `RootPOA' is a nil object reference");
	    System.exit(1);
	}

	POA root = null;
        try
        {
            root = POAHelper.narrow(obj);
        }
        catch(BAD_PARAM ex)
	{
	    System.err.println(
		"Error: `RootPOA' is not a POA object reference");
	    System.exit(1);
	}

	return root;
    }

    static boolean
    Compare(byte[] id1, byte[] id2)
    {
	//
	// TODO: efficient method to doing this?
	//
	if(id1.length != id2.length)
	    return false;
	for(int i = 0 ; i < id1.length ; i++)
	    if(id1[i] != id2[i])
		return false;
	return true;
	//return id1.equals(id2);
    }
}
